src/app/util/Product/Product.js

```diff
@@ -56,13 +56,15 @@ const getIndexedConfigurableOptions = (configurableOptions, indexedAttributes) =
     }, {})
 );
 
-const getIndexedVariants = variants => variants.map(({ product }) => {
-    const { attributes } = product;
-    return {
-        ...product,
-        attributes: getIndexedAttributes(attributes)
-    };
-});
+const getIndexedVariants = variants => variants
+    .filter(({ product }) => product !== null)
```

Comment:
Sorry for not pointing out in the first review iteration: here you once again iterate through the array twice, while filtering and while mapping, please rewrite this to reduce :)